build_runner/web/graph_viz_main.dart

```diff
@@ -2,32 +2,40 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import 'dart:async';
 import 'dart:convert';
-import 'dart:html';
-import 'dart:js' as js;
+import 'dart:js_interop';
 
-final _graphReference = js.context[r'$build']! as js.JsObject;
-final _details = document.getElementById('details')!;
+import 'package:http/http.dart' as http;
+import 'package:web/web.dart';
+
+final _graphReference = window[r'$build'] as Graph;
+final _details = document.getElementById('details')! as HTMLElement;
+
+extension type Graph(JSObject _) implements JSObject {
+  external void initializeGraph(JSFunction callback);
+
+  external void setData(JSObject data);
+}
 
 void main() async {
-  var filterBox = document.getElementById('filter') as InputElement;
-  var searchBox = document.getElementById('searchbox') as InputElement;
-  var searchForm = document.getElementById('searchform')!;
+  var filterBox = document.getElementById('filter') as HTMLInputElement;
+  var searchBox = document.getElementById('searchbox') as HTMLInputElement;
+  var searchForm = document.getElementById('searchform') as HTMLFormElement;
   searchForm.onSubmit.listen((e) {
     e.preventDefault();
-    _focus(searchBox.value!.trim(),
-        filter: filterBox.value!.isNotEmpty ? filterBox.value : null);
+    _focus(searchBox.value.trim(),
+        filterBox.value.isNotEmpty ? filterBox.value : null);
   });
-  _graphReference.callMethod('initializeGraph', [_focus]);
+  _graphReference.initializeGraph(_focus.toJS);
 }
 
 void _error(String message, [Object? error, StackTrace? stack]) {
   var msg = [message, error, stack].where((e) => e != null).join('\n');
-  _details.innerHtml = '<pre>$msg</pre>';
+  _details.innerHTML = '<pre>$msg</pre>'.toJS;
+  console.error(msg.toJS);
 }
 
-Future _focus(String query, {String? filter}) async {
+void _focus(String query, [String? filter]) async {
```

Comment:
Is the intention for this method to never be awaited (it looks like that's the case today already)? If so, it's fine to not do anything with the `Future`, but I can't tell.

Reply:
Yes, the return value is not used. This function is used in js interop which was complaining about the Future return type - but since I don't actually need the future this was the easiest solution :).

Reply:
(and to be clear the JS side also doesn't use the return value, its just a callback for an event)

Reply:
Sounds good, the switch to `void` makes sense then.